cleverpush/src/main/java/com/cleverpush/AddSubscriptionTags.java

```diff
@@ -0,0 +1,97 @@
+package com.cleverpush;
+
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import com.cleverpush.listener.AddTagCompletedListener;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class AddSubscriptionTags implements AddTagCompletedListener {
+
+    private String[] tagIds;
+    private String subscriptionId = null;
+    private String channelId;
+
+    public AddSubscriptionTags(String subscriptionId, String channelId, String... tagIds) {
+        this.subscriptionId = subscriptionId;
+        this.channelId = channelId;
+        this.tagIds = tagIds;
+    }
+
+    @Override
+    public void tagAdded(int currentPositionOfTagToAdd) {
+        if (currentPositionOfTagToAdd != tagIds.length - 1) {
+            currentPositionOfTagToAdd++;
+            addSubscriptionTag(tagIds[currentPositionOfTagToAdd], this, currentPositionOfTagToAdd);
+        }
+    }
+
+    public void addMultipleSubscriptionTags() {
+        addSubscriptionTag(tagIds[0], this, 0);
+    }
+
+    public void addSubscriptionTag() {
+        addSubscriptionTag(tagIds[0], null, -1);
+    }
+
+    public void addSubscriptionTag(String tagId, AddTagCompletedListener addTagCompletedListener, int currentPositionOfTagToAdd) {
+        if (subscriptionId != null) {
+            Set<String> tags = this.getSubscriptionTags();
+            if (tags.contains(tagId)) {
+                if (addTagCompletedListener != null) {
+                    addTagCompletedListener.tagAdded(currentPositionOfTagToAdd);
+                }
+                Log.d("CleverPush", "Subscription already has tag - skipping API call " + tagId);
+                return;
+
+            }
+
+            JSONObject jsonBody = new JSONObject();
+            try {
+                jsonBody.put("channelId", this.channelId);
+                jsonBody.put("tagId", tagId);
+                jsonBody.put("subscriptionId", subscriptionId);
+            } catch (JSONException ex) {
+                Log.e("CleverPush", ex.getMessage(), ex);
+            }
+
+            tags.add(tagId);
+            SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(CleverPush.context);
+
+            CleverPushHttpClient.post("/subscription/tag", jsonBody, new CleverPushHttpClient.ResponseHandler() {
```

Comment:
please export ResponseHandler to a method returning the it

so it looks like this
CleverPushHttpClient.post("/subscription/tag", jsonBody, httpClientResponseHandler);


Cle